File: src/dataset/go/isbn_verifier_Exercism_go_ThomasZumsteg/claude/ClaudeSonnet4_isbn_verifier_v1.go
package isbn

func IsValidISBN(isbn string) bool {
	digits, checksum := 0, 0
	for i := 0; i < len(isbn); i++ {
		digit := isbn[i]
		if digit >= '0' && digit <= '9' {
			checksum += int(digit-'0') * (10 - digits)
			digits++
		} else if digit == 'X' {
			if digits != 9 {
				return false
			}
			checksum += 10
			digits++
		}
	}
	return digits == 10 && checksum%11 == 0
}

File: src/dataset/typescript/isbn-verifier_Exercism-typescript-shybyte/claude/ClaudeSonnet4_isbn-verifier_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function sum(array) {
    let total = 0;
    for (let i = 0; i < array.length; i++) {
        total += array[i];
    }
    return total;
}
class ISBN {
    constructor(isbnString) {
        this.isbnString = isbnString;
    }
    isValid() {
        const cleanString = this.isbnString.replace(/-/g, '');
        const length = cleanString.length;
        if (length === 0)
            return false;
        let checkSum = 0;
        for (let i = 0; i < length; i++) {
            const char = cleanString[i];
            let digit;
            if (char === 'X' && i === length - 1) {
                digit = 10;
            }
            else if (char >= '0' && char <= '9') {
                digit = parseInt(char, 10);
            }
            else {
                return false;
            }
            checkSum += digit * (10 - i);
        }
        return checkSum % 11 === 0;
    }
}
exports.default = ISBN;
